core/test/mpi/distributed/solver/multigrid.cpp

```diff
@@ -0,0 +1,212 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/distributed/matrix.hpp>
+#include <ginkgo/core/distributed/partition.hpp>
+#include <ginkgo/core/distributed/vector.hpp>
+#include <ginkgo/core/solver/cg.hpp>
+#include <ginkgo/core/solver/multigrid.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+
+
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+using value_type = float;
+using local_index_type = int;
+using global_index_type = int;
+using dist_mtx_type =
+    gko::experimental::distributed::Matrix<value_type, local_index_type,
+                                           global_index_type>;
+using dist_vec_type = gko::experimental::distributed::Vector<value_type>;
+
+
+class DummyLinOp : public gko::EnableLinOp<DummyLinOp>,
+                   public gko::EnableCreateMethod<DummyLinOp> {
+public:
+    DummyLinOp(std::shared_ptr<const gko::Executor> exec,
+               gko::dim<2> size = gko::dim<2>{})
+        : EnableLinOp<DummyLinOp>(exec, size)
+    {}
+
+protected:
+    void apply_impl(const gko::LinOp* b, gko::LinOp* x) const override {}
+
+    void apply_impl(const gko::LinOp* alpha, const gko::LinOp* b,
+                    const gko::LinOp* beta, gko::LinOp* x) const override
+    {}
+};
+
+
+template <typename ValueType>
+class DummyMultigridLevelWithFactory
```

Comment:
I think this needs some documentation why it is actually needed. Also, can't the MultigridLevel implementation adapted to provide that functionality? This seems a bit much for testing purposes.

Reply:
Yes, it is entirely for the testing process. It's good to have when we do not really have the distributed multigridlevel generation at the moment. Also, the simple dummy component can simplify the test dependency, which means we do not need to rely on another maybe-failed components.